### PR TITLE
fix freshness calculations in statement distribution

### PR DESCRIPTION
**Pull request by rphmeier**
This breaks a lot of tests, even though it's now the 'correct' behavior.

### CODE REVIEW

File: node/network/statement-distribution/src/lib.rs [hunk 1]
```diff
@@ -269,7 +269,9 @@ impl PeerRelayParentKnowledge {
 			CompactStatement::Seconded(ref h) => {
 				self.seconded_counts.entry(fingerprint.1).or_default().note_local(h.clone());
 
-				self.sent_candidates.insert(h.clone())
+				let was_known = self.is_known_candidate(h);
+				self.sent_candidates.insert(h.clone());
+				was_known
```
Comment (eskimor) [inline node/network/statement-distribution/src/lib.rs:274]: Shouldn't this be !was_known ... the function is supposed to return whether the pear learned from this statement the first time, which is equivalent to it was not known by the peer.

File: node/network/statement-distribution/src/lib.rs [hunk 2]
```diff
@@ -338,14 +340,14 @@ impl PeerRelayParentKnowledge {
 					return Err(COST_UNEXPECTED_STATEMENT_REMOTE)
 				}
 
-				h
+				(h, self.is_known_candidate(h))
```
Comment (eskimor) [inline node/network/statement-distribution/src/lib.rs:343]: Same - it is fresh if it was not known.

**Comment by eskimor** [comment]
My understanding is the missing `!` is a logic error. The actual purpose of this PR is to also take into consideration candidates in `received_candidates` instead of just in `sent_candidates` - correct?

**Comment by rphmeier** [comment]
Yes, that's right. This can lead to too many messages being sent, possibly.

**Comment by eskimor** [comment]
bot merge

**Review by eskimor — APPROVED** [review]

**Pull request merged**